www/ec2/ec2.inc.php

```diff
@@ -385,68 +406,65 @@ function EC2_DeleteOrphanedVolumes() {
 
 function EC2_GetRunningInstances() {
   $now = time();
+  $regions = array();
   $instances = array();
-  $key = GetSetting('ec2_key');
-  $secret = GetSetting('ec2_secret');
-  if ($key && $secret) {
-    $locations = EC2_GetAMILocations();
-    try {
-      $ec2 = \Aws\Ec2\Ec2Client::factory(array('key' => $key, 'secret' => $secret, 'region' => 'us-east-1'));
-      $regions = array();
-      $response = $ec2->describeRegions();
-      if (isset($response['Regions'])) {
-        foreach ($response['Regions'] as $region)
-          $regions[] = $region['RegionName'];
-      }
-    } catch (\Aws\Ec2\Exception\Ec2Exception $e) {
-      $error = $e->getMessage();
-      EC2LogError("Listing running EC2 instances: $error");
+
+  $locations = EC2_GetAMILocations();
+  $ec2 = EC2_GetSdkClient();

```

Comment:
Either move this inside the try/except block or (if it is handled above), check for a null $ec2 (same for all occurrences below)